src/api/app/jobs/create_project_log_entry_job.rb

```diff
@@ -4,6 +4,6 @@ class CreateProjectLogEntryJob < ApplicationJob
   def perform(event_id)
     event = Event::Base.find(event_id)
     entry = ProjectLogEntry.create_from(event)
-    event.update_attributes(project_logged: true) if entry.persisted?
+    event.update_attributes(project_logged: true) # If the ProjectLogEntry was invalid then we still mark the event as logged
```

Comment:
Why don't you check if the entry is invalid? 

I mean, being invalid is not the same as not persisted.

Reply:
My thinking is that we should set project_logged = true regardless of whatever happens in `ProjectLogEntry.create_from(event)`. Because say the ProjectLogEntry create failed not because its invalid but because of some kind of error. Then we will just have that event sit in the backlog with project_logged=false and it will never be re-tried.

Really we should rely on delayed_jobs to handle failure and retries since it has that functionality built-in, and not rely on the events table to track this stuff. Also in the next sprint we will be getting rid of the project_logged column entirely so this code will on be here for a short time anyway..

Reply:
Ok, then if you don't mind about the `create_from` call why not to move the `update_attributes`one just before the `create_from`?

That way you won't care at all... and if it fails it will fail after marking the event at least.

Reply:
done